HelperViews/ContextMenuActions.cs

```diff
@@ -59,5 +60,70 @@ public static void SendToBackButton_Clicked(View targetElement, ContextMenu cont
                 Debug.WriteLine("SendToBack: Moved the focused view to the back.");
             }
         }
+
+        public static void CutElement(View targetElement, ContextMenu contextMenu, EventArgs e, AbsoluteLayout designerFrame)
+        {
+            if (targetElement != null)
+            {
+                ClipboardElement = targetElement;
+                designerFrame.Children.Remove(targetElement);
+                contextMenu.Close();
+            }
+        }
+
+
+        public static void CopyElement(View targetElement, ContextMenu contextMenu, EventArgs e)
+        {
+            if (targetElement != null)
+            {
+                ClipboardElement = targetElement;
+                contextMenu.Close();
+            }
+        }
+
+        public static void PasteElement(View targetElement, ContextMenu contextMenu, EventArgs e, AbsoluteLayout designerFrame)
+        {
+            if (ClipboardElement != null)
+            {
+                var newElement = CloneView(ClipboardElement);
+                designerFrame.Children.Add(newElement);
+                ClipboardElement = null;
+
+                //TODO: fix to paste where the pointer is clicked
+            }
+            contextMenu.Close();
+        }
+
+        private static View CloneView(View originalView)
+        {
+            
+            var newView = ElementCreator.Create(originalView.GetType().Name);
+
+            newView.BindingContext = originalView.BindingContext;
```

Comment:
Does this mean that they have same inherent context?

Reply:
Paste is not working perfectly. we still have to make it work.